pactum/resources.py

```diff
@@ -8,21 +11,48 @@ def __init__(self, name=None):
 class Resource(BaseResource):
     def __init__(self, name=None, fields=None, behaviors=None):
         super().__init__(name)
+
         self._mapfields = {}
+        self.fields = []
+        self._load_fields(fields)
 
+        if behaviors is None:
+            behaviors = getattr(self, "behavior", [])
+        self.behaviors = behaviors
+
+    def _load_fields(self, fields):
         if fields is None:
             fields = getattr(self.__class__, "fields", [])
 
         for field in fields:
-            if self._mapfields.get(field.name):
-                raise AttributeError("Duplicate field names")
-            self._mapfields[field.name] = field
-            field.parent = self
-        self.fields = fields
+            self._add_field(field.name, field)
```

Comment:
Estou vendo duas chamadas pra add_field dentro pra cada field dependendo se você define por `class Fields` ou port `fields = [] Se o usuário definir das duas maneiras, as duas serão aceitas. Isso está correto?

Não daria rpa agrupas os fields das duas formas de definicão num mermo `for`?

Reply:
é essa a ideia... permitir os dois modos... mas não consegui juntar os 2 porque a lógica de adição dos 2 é diferente...

Reply:
dá pra colocar as duas formas em um mesmo "formato" internamente e depois chamar o for com o add_field, não?

Reply:
dá no mesmo: 2 loops, não? :)

Reply:
tentei fazer uma demonstracão e entendi porque estão separados. Tá certo.